src/coredump/coredump.c

```diff
@@ -129,6 +131,7 @@ static const char * const meta_field_names[_META_MAX] = {
         [META_ARGV_TIMESTAMP] = "COREDUMP_TIMESTAMP=",
         [META_ARGV_RLIMIT]    = "COREDUMP_RLIMIT=",
         [META_ARGV_HOSTNAME]  = "COREDUMP_HOSTNAME=",
+        [META_ARGV_PIDFD]     = "COREDUMP_PIDFD=",
```

Comment:
i don't see the point of this... it doesn't make sense to generate a journal field from this, fds are ephemeral concepts, and in this very specific case it's going to strictly map to "3" anyway, which seems very useless to attach.

maybe keep an entry here, but assign it an explicit NULL with a comment indicating that this is set to null because we don't want to generate a journal property from it

Reply:
nah, you don't need to have a journal property for that at all. Just fill in the Context->pidref field already early, the same way as process_socket() already does that based on the pidfd received via sockets, before it calls context_parse_iovw().

i.e. make gather_pid_metadata_from_argv() more like process_socket() and already fill in the Context->pidref field before it calls context_parse_iovw().

iiuc your current code is actually quite broken inside a container if an old coredump handler from outside delivers a coredump via the socket protocol and doesn't include the pidfd. Because in that codepath you'll blindly end up using fd3 for the coredump, and that might be something completely different or even closed.